### ISSUE TITLE
Support XUnit's x86 .exe runner

### ISSUE THREAD

**Issue by martingemme** [cake-build/cake#670]
Probably something I'm missing regarding running XUnit with x86, but:

The problem I'm having is mainly when trying to run tests based on ARM architecture for Windows Phone, where the 64 bit runner doesn't work with the ARM/x86 set. However, when using `xunit.console.x86.exe`, it works. It would be nice to (solution) add a parameter to XUnit2Settings to use the x86 .exe instead of the 64 bit one (aka `xunit.console.exe`), or detect the build somehow.

Related to [this](https://github.com/cake-build/cake/blob/f44c6ae6b83f4115075a07ec2b1e3a081f4f7c5d/src/Cake.Common/Tools/XUnit/XUnit2Runner.cs#L152) ?
Thanks!

**Comment by devlead** [comment]
What you could do today is to specify the tool path:
```cake
#tool "xunit.runner.console"
XUnit2("./src/**/bin/" + configuration + "/*.Tests.dll", new XUnit2Settings {
        ToolPath = "./tools/xunit.runner.console/tools/xunit.console.x86.exe"
        });
```

**Comment by martingemme** [comment]
@devlead That works! 
Anything in the works to do something according to the build target?

Thanks a lot!

**Comment by wieslawsoltes** [comment]
Having same issue, would be nice to have in `XUnit2Settings ` parameter for target platform `x86` or `x64`.

**Comment by phillipsj** [comment]
@devlead I will take this one.  If there are any design considerations I will post back here.

**Comment by devlead** [comment]
@phillipsj perfect 👍

**Comment by phillipsj** [comment]
@devlead @gep13 @patriksvensson So I have two different implementations.  One is a tool resolver and the other changes how tooling works in general.  One addresses xUnit and the other allows all tools to define x86 support.  Is there a preference?

**Comment by patriksvensson** [comment]
@phillipsj Since all tools doesn't have a corresponding x86-implementation, I think it would suffice with a fix for xunit at this time.

**Comment by devlead** [comment]
Fixed by #1336
